sentry-android-distribution/src/main/java/io/sentry/android/distribution/DistributionIntegration.kt

```diff
@@ -86,12 +86,17 @@ public class DistributionIntegration(context: Context) : Integration, IDistribut
   /**
    * Check for available updates asynchronously using a callback.
    *
+   * Note: The callback will be invoked on a background thread. If you need to update UI or perform
+   * main-thread operations, dispatch the result to the main thread yourself.
+   *
    * @param onResult Callback that will be called with the UpdateStatus result
    */
   public override fun checkForUpdate(onResult: IDistributionApi.UpdateCallback) {
-    // TODO implement this in a async way
-    val result = checkForUpdateBlocking()
-    onResult.onResult(result)
+    Thread {
+        val result = checkForUpdateBlocking()
+        onResult.onResult(result)
```

Comment:
I think ideally we should:
- not create a new thread for this
- avoid a call back if we can (people will not read the comment and be confused when this crashes)

Looks like we already have some sort of common executor service which is used in the codebase and returns a future. Doesn't seem like it's Android aware (with Handlers / Loopers) but seems used from Android code.

- https://github.com/getsentry/sentry-java/blob/cf708bd2c180f26ddbfd25ca36bf5d3488da3046/sentry/src/main/java/io/sentry/HostnameCache.java#L133
- https://github.com/getsentry/sentry-java/blob/cf708bd2c180f26ddbfd25ca36bf5d3488da3046/sentry/src/main/java/io/sentry/SentryExecutorService.java#L71
- https://developer.android.com/develop/background-work/background-tasks/asynchronous/java-threads#use-handlers